src/plugins/visualizations/public/embeddable/visualize_embeddable_factory.tsx

```diff
@@ -133,14 +134,21 @@ export class VisualizeEmbeddableFactory
     }
   }
 
-  public async create() {
+  public async create(input: VisualizeInput & { hideVisModal: boolean }, parent?: IContainer) {
```

Comment:
could we rather say that input can also be undefined, and if its undefined we show the modal else we create from the input

Reply:
input won't be undefined in case of a "regular save", it contains timeRange and query and a few other parameters:
https://github.com/elastic/kibana/blob/5348110a469c0c4e405d28673cc7dbddfaeefe11/src/plugins/visualizations/public/embeddable/visualize_embeddable.ts#L61